crates/sui-types/src/committee.rs

```diff
@@ -46,60 +57,100 @@ pub struct Committee {
 impl Committee {
     pub fn new(
         epoch: EpochId,
-        voting_rights: BTreeMap<AuthorityName, StakeUnit>,
+        voting_weights: BTreeMap<AuthorityName, VoteUnit>,
     ) -> SuiResult<Self> {
-        let mut voting_rights: Vec<(AuthorityName, StakeUnit)> =
-            voting_rights.iter().map(|(a, s)| (*a, *s)).collect();
+        let mut voting_weights: Vec<(AuthorityName, VoteUnit)> =
+            voting_weights.iter().map(|(a, s)| (*a, *s)).collect();
 
         fp_ensure!(
             // Actual committee size is enforced in sui_system.move.
             // This is just to ensure that choose_multiple_weighted can't fail.
-            voting_rights.len() < u32::MAX.try_into().unwrap(),
+            voting_weights.len() < u32::MAX.try_into().unwrap(),
             SuiError::InvalidCommittee("committee has too many members".into())
         );
 
         fp_ensure!(
-            !voting_rights.is_empty(),
+            !voting_weights.is_empty(),
             SuiError::InvalidCommittee("committee has 0 members".into())
         );
 
         fp_ensure!(
-            voting_rights.iter().any(|(_, s)| *s != 0),
+            voting_weights.iter().any(|(_, s)| *s != 0),
             SuiError::InvalidCommittee(
-                "at least one committee member must have non-zero stake.".into()
+                "at least one committee member must have non-zero voting power.".into()
             )
         );
 
-        voting_rights.sort_by_key(|(a, _)| *a);
-        let total_votes = voting_rights.iter().map(|(_, votes)| *votes).sum();
+        voting_weights.sort_by_key(|(a, _)| *a);
+        
+        let total_votes: VoteUnit = voting_weights.iter().map(|(_, votes)| *votes).sum();
+        fp_ensure!(
+            total_votes == TOTAL_VOTING_POWER,
+            SuiError::InvalidCommittee(
+                format!("total voting power of a committee is {}, must be {}", total_votes, TOTAL_VOTING_POWER)
+            )
+        );
 
-        let (expanded_keys, index_map) = Self::load_inner(&voting_rights);
+        let (expanded_keys, index_map) = Self::load_inner(&voting_weights);
 
         Ok(Committee {
             epoch,
-            voting_rights,
-            total_votes,
+            voting_weights,
             expanded_keys,
             index_map,
             loaded: true,
         })
     }
 
+    /// Normalize the given weights to TOTAL_VOTING_POWER and create the committee
+    pub fn normalize_from_weights(
```

Comment:
Is it not possible to require all callers to set up the initial weights correctly?
I am a bit concerned that the normalization algorithm here is inconsistent with what's in Move.
There is another constraint where each voting power will be capped by 10%.

If it's really hard, let's see if we could still cap it.
We should at least suffix this function with `_for_testing` and make sure it's only called from tests.